src/Config/RuntimeConfigLoader.cs

```diff
@@ -0,0 +1,301 @@
+    }
+
+    /// <summary>
+    /// Returns the default config file name.
+    /// </summary>
+    public const string DEFAULT_CONFIG_FILE_NAME = $"{CONFIGFILE_NAME}{CONFIG_EXTENSION}";
+
+    /// <summary>
+    /// Generates the config file name and a corresponding overridden file name,
+    /// With precedence given to overridden file name, returns that name
+    /// if the file exists in the current directory, else an empty string.
+    /// </summary>
+    /// <param name="environmentValue">Name of the environment to
+    /// generate the config file name for.</param>
+    /// <param name="considerOverrides">whether to look for overrides file or not.</param>
+    /// <returns></returns>
+    private string GetFileName(string? environmentValue, bool considerOverrides)
+    {
+        string fileNameWithoutExtension = _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName);
+        string fileExtension = _fileSystem.Path.GetExtension(_baseConfigFileName);
+        string configFileName =
+            !string.IsNullOrEmpty(environmentValue)
+            ? $"{fileNameWithoutExtension}.{environmentValue}"
+            : $"{fileNameWithoutExtension}";
+        string configFileNameWithExtension = $"{configFileName}{fileExtension}";
+        string overriddenConfigFileNameWithExtension = GetOverriddenName(configFileName);
+
+        if (considerOverrides && DoesFileExistInCurrentDirectory(overriddenConfigFileNameWithExtension))
+        {
+            return overriddenConfigFileNameWithExtension;
+        }
+
+        if (DoesFileExistInCurrentDirectory(configFileNameWithExtension))
+        {
+            return configFileNameWithExtension;
+        }
+
+        return string.Empty;
+    }
+
+    private static string GetOverriddenName(string fileName)
+    {
+        return $"{fileName}.overrides{CONFIG_EXTENSION}";
+    }
+
+    public bool DoesFileExistInCurrentDirectory(string fileName)
+    {
+        string currentDir = _fileSystem.Directory.GetCurrentDirectory();
+        // Unable to use ILogger because this code is invoked before LoggerFactory
+        // is instantiated.
+        if (_fileSystem.File.Exists(_fileSystem.Path.Combine(currentDir, fileName)))
+        {
+            // This config file is logged as being found, but may not actually be used!
+            Console.WriteLine($"Found config file: {fileName}.");
+            return true;
+        }
+        else
+        {
+            // Unable to use ILogger because this code is invoked before LoggerFactory
+            // is instantiated.
+            Console.WriteLine($"Unable to find config file: {fileName} does not exist.");
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// This method reads the dab.draft.schema.json which contains the link for online published
+    /// schema for dab, based on the version of dab being used to generate the runtime config.
+    /// </summary>
+    public string GetPublishedDraftSchemaLink()
```

Comment:
I kind of think we're better off either falling back to an `aka.ms` link rather than an error - after all, not getting the schema isn't really that big a deal is it?

Reply:
Understood, more of an IDE/tooling validation of schema. Our parsing would then be the final arbiter of whether the config is good or not :P